src/sagemaker/model.py

```diff
@@ -274,9 +274,17 @@ def _compilation_image_uri(self, region, target_instance_type, framework, framew
             framework (str): The framework name.
             framework_version (str): The framework version.
         """
-        framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+        framework_prefix = ""
+        framework_suffix = ""
+        
+        if framework == "xgboost":
+            framework_suffix = "-neo"
+        else:
+            framework_prefix = "inferentia-" if target_instance_type.startswith("ml_inf") else "neo-"
+
+
         return image_uris.retrieve(
-            "{}{}".format(framework_prefix, framework),
+            "{}{}".format(framework_prefix, framework, framework_suffix),
```

Comment:
I don't think this does what you want it to do

```
>>> "{}{}".format("",'2','3')
'2'
```

Reply:
Addressed.